packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ProjectController.java

```diff
@@ -502,23 +532,54 @@ public ResponseEntity<Project> updateProject(@PathVariable("id") final UUID id,
 	@Secured(Roles.USER)
 	public ResponseEntity<Project> copyProject(@PathVariable("id") final UUID id) {
 		projectService.checkPermissionCanRead(currentUserService.get().getId(), id);
+		// time the progress takes to reach each subsequent half.
+		final Double HALFTIME_SECONDS = 2.0;
+
+		Future<Project> project;
+		Project clonedProject;
 
 		final String userId = currentUserService.get().getId();
 		final String userName = userService.getById(userId).getName();
 
-		Project project;
 		try {
-			final ProjectExport export = cloneService.exportProject(id);
-			export.getProject().setName("Copy of " + export.getProject().getName());
+			final NotificationGroupInstance<Properties> notificationInterface = new NotificationGroupInstance<Properties>(
+				clientEventService,
+				notificationService,
+				ClientEventType.CLONE_PROJECT,
+				null,
+				new Properties(id),
+				HALFTIME_SECONDS
+			);
 
-			project = cloneService.importProject(userId, userName, export);
+			notificationInterface.sendMessage("Cloning the Project...");
+
+			project = executor.submit(() -> {
+				log.info("Staring Cloning Process...");
+				final ProjectExport export = cloneService.exportProject(id);
+				export.getProject().setName("Copy of " + export.getProject().getName());
+				log.info("Cloning...");
+				Project cloneProject = cloneService.importProject(userId, userName, export);
+				log.info("Cloned...");
+				return cloneProject;
+			});
+			notificationInterface.sendFinalMessage("Cloning complete");
+			clonedProject = project.get();
+		} catch (ExecutionException e) {
+			log.error("Execution Exception exporting project", e);
+			throw new ResponseStatusException(HttpStatus.SERVICE_UNAVAILABLE, messages.get("postgres.service-unavailable"));
+		} catch (CancellationException e) {
+			log.error("Canceled exporting project", e);
+			throw new ResponseStatusException(HttpStatus.SERVICE_UNAVAILABLE, messages.get("postgres.service-unavailable"));
```

Comment:
Pedantic, but, is a `cancellationException` caused by the user cancelling something or that there was a problem trying to cancel? If its the former we probably don't want to throw an exception (or at least, a postgres error for the front end?